File: VB/dxSample_IAnyControl/UserControl_GridWIthChart.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraCharts
Imports DevExpress.XtraEditors.CustomEditor
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.Grid

Namespace dxSample_IAnyControl.IAny_Files
	Partial Public Class UserControl_GridWithChart
		Inherits UserControl

		Private ReadOnly Property ChartSeries() As SeriesBase
			Get
				If chartControl1.Series.Count = 0 Then
					CreateChartSeries(chartControl1)
				End If
				Return chartControl1.Series("PieSeries")
			End Get
		End Property
		Private Sub CreateChartSeries(ByVal chart As ChartControl)
			chart.Series.Add("PieSeries", ViewType.Pie)
		End Sub
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub UC_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			gridControl1.DataSource = DataHelper.GetData()
			InitChartSeries(ChartSeries)
			CreateChartEdit(colSales, chartControl1)
		End Sub
		Private Sub InitChartSeries(ByVal chartSeries As SeriesBase)
			If chartSeries Is Nothing Then
				Return
			End If
			Me.ChartSeries.LegendTextPattern = "{A}"
			Me.ChartSeries.ArgumentDataMember = "CategoryName"
			Me.ChartSeries.ValueDataMembers.AddRange("Quantity")
		End Sub
		Private Shared Sub CreateChartEdit(ByVal column As GridColumn, ByVal chartControl As ChartControl)
			If column.ColumnEdit IsNot Nothing Then
				Return
			End If
			Dim item As New RepositoryItemAnyControl()
			item.Control = New EmbeddedChartControl(chartControl)
			column.View.GridControl.RepositoryItems.Add(item)
			CType(column.View, GridView).OptionsSelection.EnableAppearanceHideSelection = False
			column.OptionsColumn.AllowEdit = False
			column.OptionsFilter.AllowFilter = False
			column.OptionsColumn.AllowGroup = DevExpress.Utils.DefaultBoolean.False
			column.OptionsColumn.AllowSort = DevExpress.Utils.DefaultBoolean.False
			column.ColumnEdit = item
		End Sub
	End Class
End Namespace

File: CS/dxSample_IAnyControl/UserControl_GridWIthChart.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraCharts;
using DevExpress.XtraEditors.CustomEditor;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.Grid;

namespace dxSample_IAnyControl.IAny_Files {
    public partial class UserControl_GridWithChart : UserControl {
        private SeriesBase ChartSeries {
            get {
                if (chartControl1.Series.Count == 0) {
                    CreateChartSeries(chartControl1);
                }
                return chartControl1.Series["PieSeries"];
            }
        }
        private void CreateChartSeries(ChartControl chart) {
            chart.Series.Add("PieSeries", ViewType.Pie);
        }
        public UserControl_GridWithChart() {
            InitializeComponent();
        }
        private void UC_Load(object sender, EventArgs e) {
            gridControl1.DataSource = DataHelper.GetData();
            InitChartSeries(ChartSeries);
            CreateChartEdit(colSales, chartControl1);
        }
        private void InitChartSeries(SeriesBase chartSeries) {
            if (chartSeries == null) {
                return;
            }
            ChartSeries.LegendTextPattern = "{A}";
            ChartSeries.ArgumentDataMember = "CategoryName";
            ChartSeries.ValueDataMembers.AddRange("Quantity");
        }
        private static void CreateChartEdit(GridColumn column, ChartControl chartControl) {
            if (column.ColumnEdit != null) {
                return;
            }
            RepositoryItemAnyControl item = new RepositoryItemAnyControl();
            item.Control = new EmbeddedChartControl(chartControl);
            column.View.GridControl.RepositoryItems.Add(item);
            ((GridView)column.View).OptionsSelection.EnableAppearanceHideSelection = false;
            column.OptionsColumn.AllowEdit = false;
            column.OptionsFilter.AllowFilter = false;
            column.OptionsColumn.AllowGroup = DevExpress.Utils.DefaultBoolean.False;
            column.OptionsColumn.AllowSort = DevExpress.Utils.DefaultBoolean.False;
            column.ColumnEdit = item;
        }
    }
}
